common/trace/src/trace.rs

```diff
@@ -390,10 +392,28 @@ fn abbreviate_crate(name: &str) -> &str {
     }
 }
 
-/// Sets diem trace config
+/// Sets diem trace config.
+///
+/// This should only be called once.
 pub fn set_diem_trace(config: &HashMap<String, String>) -> Result<()> {
-    SAMPLING_CONFIG.get_or_try_init(|| parse_sampling_config(config))?;
```

Comment:
Couldn't we parse the config upfront and then use `OnceCell::set` in order to have the desired behavior without needing the extra atomic?

Reply:
See the update I just made, I think it simplifies this considerably. Thanks for the review :)